### DISCUSSION TITLE
Validate only if field is provided with any value

### DISCUSSION THREAD

**Discussion by SankaranarayananMurugan** [react-hook-form/react-hook-form#2957] [Q&A]
I have a react hook form with around 60 fields, which triggers validation onChange. 
I have Submit button, which is expected to perform the complete form validation before submit, where around 50 fields are required. So far I had no issues in implementing the above.

Now I want to have Save button, which is expected to perform required validation only for around 10 fields, rest of the fields should be validated only if it is provided any value to prevent from saving invalid data. So basically, Save button is used to save an incomplete form with minimum required fields, so that it can be reopened anytime to fill the remaining fields and Submit.
 
I'm finding difficult to implement this change in validation behavior on two different actions.Can anyone please help here?

**Comment by bluebill1049** [comment]
i think you want to use `valdiate` from `register` function and combined with `getValues()`

**Reply by SankaranarayananMurugan** [reply]
Thanks @bluebill1049 , I didnt expect such a faster response.

I use Controller to render the fields, which takes care of the register/unregister. So do I have to unregister & register again manually with different set of validation based on the button (Save or Submit) being clicked?

**Reply by SankaranarayananMurugan** [reply]
I'm thinking if there is a way to trigger all validations (except required) on onChange, and perform selective required validation based on the button being clicked, as until user clicks an action button they will not be aware which fields are required.
